jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/ArtifactUtil.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Shivang Mishra - Google Summer of Code 2023 - Improve JKube UX
+ */
+package org.eclipse.jkube.kit.common.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.math.BigInteger;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+import org.eclipse.jkube.kit.common.PrefixedLogger;
+
+public class ArtifactUtil {
+  protected static final String CHECKSUM_FILENAME = ".jkube-checksum";
+
+  // this is a utility class, so no instances allowed
+  private ArtifactUtil() {
+
+  }
+
+  public static boolean isArtifactSameAsPrevious(File sourceFile, PrefixedLogger log)
+      throws IOException, NoSuchAlgorithmException {
+    Path path = sourceFile.getParentFile().toPath();
```

Comment:
I think the checksum file should be in the target folder

Reply:
It is already in the target folder. I don't know why sourceFile's parent directory gives me the tarrget folder. I thjnk sourceFile is an incorrectly named variable. 
`final File sourceFile = Objects.requireNonNull(JKubeProjectUtil.getFinalOutputArtifact(getProject())`

this is how sourceFile is initialized.